### PR TITLE
add faillible conversion

### PR DESCRIPTION
**Pull request by antoine-de**
libgeos is quite sensitive to error (when not happy libgeos crashes
index )

So we wrap the rust-geo -> geos conversion with `TryFrom` (instead of `From`) to be able to return an error before the geos crash.

We'll need to add additional checks as geos is quite sensitive (and it's not easy to know what will make it crash)

I also removed some memory leak in the binding.

Note: this breaks retrocompatibility as I removed moved some methods and also removed the `WKB` conversion method as they are not used and they were leaking.

If needed we'll need to reintroduce them, but with cleaner code

### CODE REVIEW

File: src/ffi.rs [hunk 1]
```diff
@@ -379,139 +403,211 @@ impl GGeom {
...
+            // so the old one needs to forget their c ptr to avoid double cleanup
+            external_c_obj.release();
+            for mut i in interiors_ptr {
+                i.release();
+            }
+            Ok(GGeom::new_from_c_obj(t))
+        }
+    }
+
+    pub fn create_mulipolygon(polygons: Vec<GGeom>) -> GeosResult<GGeom> {
```
Comment (Tristramg) [inline src/ffi.rs]: mulTipoligyon

File: src/from_geo.rs [hunk 2]
```diff
@@ -1,69 +1,91 @@
...
+    type Err;
+    fn try_into(self) -> Result<T, Self::Err>;
+}
+
+fn create_coord_seq<'a>(points: &'a Vec<Point<f64>>) -> CoordSeq {
+    let nb_pts = points.len();
+    let coord_seq = CoordSeq::new(nb_pts as u32, 2);
+    for i in 0..nb_pts {
+        let j = i as u32;
+        coord_seq.set_x(j, points[i].x());
```
Comment (Tristramg) [inline src/from_geo.rs]: Why `j` if it’s equal to i?

File: src/from_geo.rs [hunk 3]
```diff
@@ -1,69 +1,91 @@
...
+    type Err = Error;
+
+    fn try_into(self) -> Result<GGeom, Self::Err> {
+        let coord_seq = create_coord_seq(&self.0);
+
+        Ok(GGeom::create_line_string(coord_seq))
+    }
+}
+
+// rust geo does not have the distinction LineString/LineRing, so we create a wrapper
```
Comment (Tristramg) [inline src/from_geo.rs]: at some point, shouldn’t it be added to rust geo?

File: Cargo.toml [hunk 4]
```diff
@@ -1,11 +1,13 @@
 [package]
 name = "geos"
-version = "0.0.1"
-authors = ["<matthieu.viry@cnrs.fr>"]
+version = "0.2.0"
+authors = ["<matthieu.viry@cnrs.fr>", "Adrien Matissart <a.matissart@qwantresearch.com>", "Antoine Desbordes <antoine.desbordes@gmail.com"]
 license = "MIT"
-repository = "https://github.com/mthh/rust-geos"
+repository = "https://github.com/QwantResearch/rust-geos"
```
[the remark this answers is not in the document]
Comment (antoine-de) [reply Cargo.toml]: the crate is not yet on crates.io and the author drop the maintenance of the crates, so I think it's ok

I want to ask https://github.com/georust if they can host the project in their organization and bring visibility to this project.

The problem is that I'm not confident at all of this binding :confused:

File: src/from_geo.rs (hunk 3 above)
Comment (antoine-de) [reply src/from_geo.rs]: I think so but I'm not motivated enough to work on this :confused:

File: src/from_geo.rs (hunk 2 above)
Comment (antoine-de) [reply src/from_geo.rs]: because `j` is `u32` and i is `usize`

**Pull request merged**